src/db/players.ts

```diff
@@ -1,26 +1,73 @@
-import { Player, PrismaClient } from "@prisma/client";
+import { BankAccount, Player, PrismaClient, Role } from "@prisma/client";
 import {
   PlainPlayerResponse,
   PlayerResponse,
   RoledPlayer,
 } from "@/types/response/players";
 import {
+  OrderBy,
   PlayerRequest,
   PlayerUpdatableProps,
   getPlayerId,
 } from "@/types/request/players";
 import { parsePlayer } from "@/utils/parser";
 import CONFIG from "@/config";
+import { ForbiddenError } from "@/helpers/error";
 
 const prisma = new PrismaClient();
 
 export class PlayersDAO {
+  static get count() {
+    try {
+      return prisma.player.count();
+    } catch (e) {
+      throw e;
+    } finally {
+      prisma.$disconnect();
+    }
+  }
+
+  static _getAll = async (
```

Comment:
**suggestion (performance):** New pagination and search methods need performance optimization

The addition of pagination and search functionality is good for scalability. However, ensure that these queries are optimized for performance, especially for large datasets. Consider adding indexes on frequently searched fields and test the performance with a large number of records.